### ISSUE TITLE
0.11B not rebalancing because 0.10 spent lots of money rebalancing

### ISSUE THREAD

**Issue by sjrummins** [ksedgwic/clboss#97]
I appreciate 0.11B isn't an official release yet, but I really wanted to give it a try as 0.10 seemed to be spending way too much money rebalancing channels. (I picked 0.11B rather than the latest code on master because it had at least been tagged as a candidate release so it felt safer.)

0.11B seems to work well and isn't rebalancing anywhere near as much, but because 0.10 spent so much money rebalancing channels I see messages like this in the log:
```
EarningsRebalancer: Node xxx has bad net income (xxx), refusing to throw good money after bad
```
This might be OK, but I wonder if this means 0.11B is blaming individual nodes for 0.10's prodigality and as a result is going to make these channels less useful than they should be by refusing to rebalance them at all. Should there be some way for 0.11B to ignore money (or at least some money) spent by 0.10 to maintain channels so things start off "fair"?

Is there any way I can reset the "money spent" records for each of my channels to 0? Should I do this even if there is such a way?

(Incidentally, I am very impressed with clboss on the whole. The idea of setting up and managing a lightning node and getting inbound liquidity all seemed like a horrific challenge and it was only the existence of clboss that encouraged me to give it a try at all. A lot of what I'd read made running a node sound a very "social" activity, asking people for inbound liquidity as a kind of favour and risking upsetting them if your node doesn't cooperate well with theirs and so forth. With clboss I don't have to worry about all the warm and fuzzy social relationship stuff. :-) Thanks very much for creating it!)

**Comment by felivix** [comment]
Yes, me too, I am very impressed with clboss on the whole. Managing a lightning node and getting inbound liquidity became automatic.

However, since 21. February we do not see transactions going through our node (except yesterday).

![image](https://user-images.githubusercontent.com/47902312/156555050-0a24ff7d-a197-4106-8ae4-1c85c1590b47.png)

**Comment by ghost** [comment]
> Yes, me too, I am very impressed with clboss on the whole. Managing a lightning node and getting inbound liquidity became automatic.
> 
> However, since 21. February we do not see transactions going through our node (except yesterday).
> 
> ![image](https://user-images.githubusercontent.com/47902312/156555050-0a24ff7d-a197-4106-8ae4-1c85c1590b47.png)

Funny, I thought I was the only one.. there were no routings on my CLBOSS-managed node for a few weeks now. Not even my free Sats in my wallet are being allocated to new channels. I am wondering what happened?

**Comment by ZmnSCPxj** [comment]
Hmm. Let me propose a command like the below?

    clboss-discount-rebalance ${PEERID} ${PERCENT}

Change how much has been spent on rebalancing.

WARNING: This changes the records of CLBOSS permanently, meaning that if you use it for accounting how much you earn vs. lost, it will no longer be accurate.

---

I recently reset my node and completely destroyed the old one, and I still get this "will not throw good money after bad" thing after a while.  `InitialRealancer` still takes a lot.  Most likely I need to actually improve the rebalancer, or at least have `InitialRebalancer` choose destination that is nearby and therefore cheap (especially since `ChannelFinderByEarnedFee` is likely to propose nodes near to existing ones anyway).

**Comment by devastgh** [comment]
I was thinking about this a bit, so i'll just leave my 2c here:
I think initialrebalancer as an idea is mostly false economically.

situation a:
- you open a channel, liquidity is on your side
- clboss and you have no information about the channel behaviour, can be a sink, or a source
- what i would do in this context is: set fees to 0/0 or something low and inspect.
- if there is movement react fast and increase fees
- If there is no movent, that is an information by itself, as setting fees higher than that is pointless at this time, and the expected outgoing routing income will probably be negligible on this channel.

situation b:
- a channel is opened to you, liquidity is on their side
- clboss and you have no information about the channel behaviour
- you have no way of influencing traffic on your side with your fees
- probably shoud set high fees, so you don't get a zillion failed forwards
- what i would do in this context is just inspect and wait, as throwing money into the total unknown is not a good idea imo
- You can also inspect your peers fees. Is it high or low? 
- If it's high thats no wonder theres no traffic... why should you thrown a good sum for the balance? It's their liqudity, their fees, you run your node for your economic success not for theirs.
- If it's low and still no traffic, that can mean a lot of things, like your other outgoing fees are high, you' or your peer is not connected enough etc...

Long story short, i think active rebalancing has its uses, but we should be relatively sure about the economic gain for doing it, before actually committing. I'm much more in favour of inciting traffic with the fees, so you get "free" rebalancing or at least information about the channel behaviour.
